packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -4352,21 +4353,40 @@ addEventListener('fetch', event => {});`
 	});
 
 	describe("--experimental-assets", () => {
-		it("should not require entry point if using --experimental-assets", async () => {
+		it("should not require entry point if using --experimental-assets and should not upload any worker", async () => {
 			const assets = [
 				{ filePath: "file-1.txt", content: "Content of file-1" },
 				{ filePath: "file-2.txt", content: "Content of file-2" },
 			];
 			writeAssets(assets);
-			writeWranglerToml({
-				experimental_assets: { directory: "assets" },
-			});
-			writeWorkerSource();
+			// note that wrangler.toml and worker source are not created
+			// pass expected formBody in
 			mockUploadWorkerRequest({
-				expectedMainModule: "no-op-assets-worker.js",
+				expectedExperimentalAssetsForm: [
```

Comment:
Just a nit about naming/shapes:

This looks like a bit of a weird shape to me, but that's maybe just me not being familiar with other test util things here. `assets` is always in `metadata` and the key is always `assets`, so I'd either expect the `expectedExperimentalAssetsForm` thing to have that built in, _or_ for this to be a more generic utility that helps you validate the contents of `metadata` (e.g. `expectedMetadata`). It's not super clear in it's current state if the test would pass if there was other unexpected stuff in metadata, for example.

Reply:
Yeah I agree its a bit dodgy - a generic expectedMetadata seems like a better idea. will add that in instead